### ISSUE TITLE
Hyper CLI script fails to load HTML on Windows

### ISSUE THREAD

**Issue by Lukasa** [Lukasa/hyper#87]
Observe:

```
C:\Users\cb2\Documents\hyper>hyper GET https://http2bin.org/html
Traceback (most recent call last):
  File "C:\Python34\Scripts\hyper-script.py", line 9, in <module>
    load_entry_point('hyper==0.2.0', 'console_scripts', 'hyper')()
  File "c:\users\cb2\documents\hyper\hyper\cli.py", line 227, in main
    print(request(args))
  File "C:\Python34\lib\encodings\cp850.py", line 19, in encode
    return codecs.charmap_encode(input,self.errors,encoding_map)[0]
UnicodeEncodeError: 'charmap' codec can't encode character '\u2014' in position
1266: character maps to <undefined>
```

**Comment by Lukasa** [comment]
Note that this *may* be a Windows-only problem.

**Comment by t2y** [comment]
Yes. I could run the same command on OS X.

**Comment by Lukasa** [comment]
Cool, that makes narrowing this fix down easier. It's almost certainly because of this:

```python
>>> locale.getpreferredencoding()
'cp1252'
```

**Comment by t2y** [comment]
The response data would be encoded with content-type charset. In this case, maybe `utf-8`, I guess your terminal (command prompt?) cannot output `utf-8` encoded string?

```python
def get_content_type_and_charset(response):                                         
    charset = 'utf-8'                                                               
    content_type = response.getheader('content-type')                               
    if content_type is None:                                                        
        return 'unknown', charset                                                   
                                                                                    
    content_type = content_type.lower()                                             
    type_and_charset = content_type.split(';', 1)                                   
    ctype = type_and_charset[0].strip()                                             
    if len(type_and_charset) == 2:                                                  
        charset = type_and_charset[1].strip().split('=')[1]                         
                                                                                    
    return ctype, charset
```

**Comment by Lukasa** [comment]
Correct, Windows cannot, at least not by default. See [this bug](http://bugs.python.org/issue1602).

**Comment by t2y** [comment]
Ah, sorry, I misunderstood reading the code. That's not encode, but decode with `utf-8`. So, the response data would be decoded as Unicode string. hmm,,,

**Comment by t2y** [comment]
> [issue 1602](http://bugs.python.org/issue1602)

I see. That's a `print` function bug.

**Comment by Lukasa** [comment]
Indeed. So the question becomes, how do we handle printing on Windows?

**Comment by t2y** [comment]
I guess we can handle using `sys.stdout` with encoded string, instead of `print()` function. Could you try this patch on your Windows?

```diff
diff --git a/hyper/cli.py b/hyper/cli.py
index 3dccfca..7727e21 100644
--- a/hyper/cli.py
+++ b/hyper/cli.py
@@ -6,6 +6,7 @@ hyper/cli
 Command line interface for Hyper inspired by Httpie.
 """
 import json
+import locale
 import logging
 import sys
 from argparse import ArgumentParser, RawTextHelpFormatter
@@ -224,7 +225,10 @@ def request(args):
 def main(argv=None):
     args = parse_argument(argv)
     log.debug('Commandline Argument: %s', args)
-    print(request(args))
+    data = request(args)
+    encoding = locale.getpreferredencoding()
+    sys.stdout.buffer.write(data.encode(encoding))
+    sys.stdout.buffer.flush()
```

**Comment by t2y** [comment]
I guess `locale.getpreferredencoding()` is the same as sys.getfilesystemencoding() in almost every case. However, I'm not sure which is appropriate.

```python
FILESYSTEM_ENCODING = sys.getfilesystemencoding()                                
```

**Comment by Lukasa** [comment]
`locale.getpreferredencoding()` should be preferred, I think.

**Comment by Lukasa** [comment]
That patch works fine on Window for the /html endpoint. It does not work for /encoding/utf8, but updating the `data.encode(encoding)` call to read `data.encode(encoding, errors='replace')` fixes that too.

**Comment by t2y** [comment]
OK. If so, it's also better to use `locale.getpreferredencoding()` for request data, instead of `FILESYSTEM_ENCODING`.

**Comment by t2y** [comment]
Thank you for confirming. I'm going to fix and send PR.

**Comment by Lukasa** [comment]
Brilliant, thanks. =)

**Comment by Lukasa** [comment]
Resolved by #88.
